### DISCUSSION TITLE
The docker container is throwing connection error if host machine is also running a mongodb on port 27017.

### DISCUSSION THREAD

**Discussion by akdasUAF** [docker-library/mongo#688] [General]
**$ sudo docker run -d -p 27017:27018 --name=mongo-example mongo:latest**
Unable to find image 'mongo:latest' locally
latest: Pulling from library/mongo
Digest: sha256:54f906f0ed3092d65237ff01d22023f43ad60a6872be98234aca24abcbabac14
Status: Downloaded newer image for mongo:latest
d23655566f7548a1c6ce706776d45a889ca03cc393da4208c83ad0d55cb3fe66
docker: Error response from daemon: driver failed programming external connectivity on endpoint mongo-example (e7c7b331af185963952ed4e978e6164f489ffc0fdc8da721d35bd127de9987b3): Error starting userland proxy: listen tcp4 0.0.0.0:27017: bind: address already in use.

**Comment by tianon** [comment]
`-p 27017:27018` this is backwards (this is trying to map port 27018 in the container to 27017 on your host, which is why you're getting a bind error)

**Comment by whalelines** [comment]
This is expected as you cannot have two services listening on the same port. If you want to run two different instances of mongodb, they will have to listen on different ports.

Perhaps you just have your ports swapped in your `docker run` command? The host port is first and the container port is second. Did you mean `sudo docker run -d -p 27018:27017 --name=mongo-example mongo:latest`.

https://docs.docker.com/reference/cli/docker/container/run/#publish
